### PR TITLE
spearbit 37, 61: use SafeCast and remove unnecessary casting

### PR DESCRIPTION
**Pull request by haoze-grvt**
- Use SafeCast functions to replace unsafe downcasting
- Remove unnecessary castings
- Remove unnecessary comment & function

### CODE REVIEW

**Review by minhbsq — APPROVED** [review]
LGTM

File: contracts/exchange/api/ConfigContract.sol [hunk 1]
```diff
@@ -59,43 +59,52 @@ contract ConfigContract is BaseContract {
   ///////////////////////////////////////////////////////////////////
   /// Config Accessors
   ///////////////////////////////////////////////////////////////////
+
+  // unsafe casting here is expected, as the byte32 value represents an signed integer
   function _configToInt(bytes32 v) internal pure returns (int64) {
-    return int64(uint64(uint256(v)));
+    return int64(SafeCast.toUint64(uint256(v)));
```
Comment (minhbsq) [inline contracts/exchange/api/ConfigContract.sol:65]: shouldn't this bee `SafeCast.toInt64(SafeCast.toInt256(uint256(v))))`

Comment (haoze-grvt) [reply contracts/exchange/api/ConfigContract.sol:65]: No. We encode int with only the lower 8 bytes of the `bytes32` value. If we do `SafeCast.toInt64(SafeCast.toInt256(uint256(v))))`, `SafeCast.toInt64` would overflow for negative `int64` values(highest bit set to 1).

Comment (haoze-grvt) [reply contracts/exchange/api/ConfigContract.sol:65]: The centiBeep config is a better example
```
  // unsafe casting here is expected, as the byte32 value represents an signed integer
  function _centiBeepToConfig(int32 v) internal pure returns (bytes32) {
    return bytes32(uint256(uint32(v)));
  }

  // unsafe casting here is expected, as the byte32 value represents an signed integer
  function _configToCentibeep(bytes32 v) internal pure returns (int32) {
    return int32(SafeCast.toUint32(uint256(v)));
  }
```

Comment (minhbsq) [reply contracts/exchange/api/ConfigContract.sol:65]: ah that makes sense!

**Review by minhbsq — APPROVED** [review]
lgtm!

**Pull request merged**